tests/store/artifact/test_databricks_artifact_repo.py

```diff
@@ -274,6 +275,132 @@ def test_log_artifact_azure_blob_client_sas_error(self, databricks_artifact_repo
+            databricks_artifact_repo.log_artifact(test_file.strpath, artifact_path)
+            write_credential_infos_mock.assert_called_with(
+                run_id=MOCK_RUN_ID, paths=[expected_location]
+            )
+            # test with block size 5
+            request_mock.assert_any_call(
+                "put",
+                MOCK_ADLS_GEN2_SIGNED_URI + "?resource=file",
+                headers=filtered_azure_headers,
+            )
+            request_mock.assert_any_call(
+                "patch",
+                MOCK_ADLS_GEN2_SIGNED_URI + "?action=append&position=0",
+                data=ANY,
+                headers=filtered_azure_headers,
+            )
+            request_mock.assert_any_call(
+                "patch",
+                MOCK_ADLS_GEN2_SIGNED_URI + "?action=append&position=5",
+                data=ANY,
+                headers=filtered_azure_headers,
+            )
+            request_mock.assert_any_call(
+                "patch",
+                MOCK_ADLS_GEN2_SIGNED_URI + "?action=append&position=10",
+                data=ANY,
+                headers=filtered_azure_headers,
+            )
+            request_mock.assert_called_with(
+                "patch",
+                MOCK_ADLS_GEN2_SIGNED_URI + "?action=flush&position=14",
+                headers=filtered_azure_headers,
+            )
+
+    def test_log_artifact_adls_gen2_flush_error(self, databricks_artifact_repo, test_file):
+        mock_successful_response = Response()
+        mock_successful_response.status_code = 200
+        mock_successful_response.close = lambda: None
+        mock_error_response = MlflowException("MOCK ERROR")
+        with mock.patch(
+            DATABRICKS_ARTIFACT_REPOSITORY + "._get_write_credential_infos"
+        ) as write_credential_infos_mock, mock.patch(
+            "mlflow.utils.rest_utils.cloud_storage_http_request",
+            side_effect=[mock_successful_response, mock_error_response],
+        ) as request_mock:
+            mock_credential_info = ArtifactCredentialInfo(
+                signed_uri=MOCK_ADLS_GEN2_SIGNED_URI,
+                type=ArtifactCredentialType.AZURE_ADLS_GEN2_SAS_URI,
+            )
+            write_credential_infos_mock.return_value = [mock_credential_info]
+            with pytest.raises(MlflowException, match=r"MOCK ERROR"):
+                databricks_artifact_repo.log_artifact(test_file.strpath)
+            write_credential_infos_mock.assert_called_with(run_id=MOCK_RUN_ID, paths=ANY)
+            request_mock.assert_any_call(
+                "put",
+                MOCK_ADLS_GEN2_SIGNED_URI + "?resource=file",
+                headers={},
+            )
+            # test with azure max block size
+            request_mock.assert_any_call(
+                "patch",
+                MOCK_ADLS_GEN2_SIGNED_URI + "?action=append&position=0&flush=true",
+                data=ANY,
+                headers={},
+            )
```

Comment:
```suggestion
            assert request_mock.mock_calls == [
                mock.call(
                    "put",
                    MOCK_ADLS_GEN2_SIGNED_URI + "?resource=file",
                    headers={},
                ),
                # test with azure max block size
                mock.call(
                    "patch",
                    MOCK_ADLS_GEN2_SIGNED_URI + "?action=append&position=0&flush=true",
                    data=ANY,
                    headers={},
                ),
            ]
```

to ensure the number of requests and order.

Reply:
Will change